site/frontend/src/pages/compare/artifact-size/artifact-size-table.vue

```diff
@@ -0,0 +1,152 @@
+<script setup lang="ts">
+import {ArtifactDescription} from "../types";
+import {diffClass, formatSize} from "../shared";
+
+const props = defineProps<{
+  a: ArtifactDescription;
+  b: ArtifactDescription;
+}>();
+
+// Sort binaries first, libraries later. Then within each category, sort in descending order by size
+// of the original (a) artifact component.
+const components = Object.entries(props.a.component_sizes)
+  .sort((a, b) => {
+    const aLib = a[0].startsWith("lib");
+    const bLib = b[0].startsWith("lib");
+    if (aLib && !bLib) {
+      return 1;
+    } else if (!aLib && bLib) {
+      return -1;
+    } else {
+      return b[1] - a[1];
+    }
+  })
+  .map((item) => item[0]);
+
+function formatName(component: string): string {
+  if (component.startsWith("lib")) {
+    return `${component}.so`;
+  }
+  return component;
+}
+
+function formatValue(value: number | undefined): string {
+  if (value === undefined) {
+    return "-";
+  }
+  return formatSize(value);
+}
+
+function formatChange(a: number | undefined, b: number | undefined): string {
+  if (a === undefined || b === undefined) {
+    return "-";
+  }
+  const diff = b - a;
+  const formatted = formatSize(Math.abs(diff));
+  if (diff < 0) {
+    return `-${formatted}`;
+  }
+  return formatted;
+}
+
+function formatPercentChange(
+  a: number | undefined,
+  b: number | undefined
+): string {
+  if (a === undefined || b === undefined) {
+    return "-";
+  }
+  return `${(((b - a) / a) * 100).toFixed(3)}%`;
+}
+
+function getClass(a: number | undefined, b: number | undefined): string {
+  if (a === undefined || b === undefined || a == b) {
+    return "";
+  }
+  return diffClass(b - a);
+}
+</script>
+
+<template>
+  <div class="category-title">Artifact component sizes</div>
+  <table>
+    <thead>
+      <tr>
+        <th>Component</th>
+        <th>Before</th>
+        <th>After</th>
+        <th>Change</th>
+        <th>% Change</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr v-for="component in components">
+        <td class="component">{{ formatName(component) }}</td>
+        <td>
+          <div class="aligned">
+            {{ formatValue(a.component_sizes[component]) }}
```

Comment:
since these values are formatted and rounded, it could be nice to display the full value on a tooltip: we may have no difference in the before/after columns, lost in the rounding after the 2nd digit, but some size difference in the "change" column.

Reply:
Good point, added titles.